### PR TITLE
Make the App's title optional on web

### PR DESCRIPTION
**Pull request by tugorez**
Title (in web) results in updating the [title element][1] which is  a global property. This is problematic in embedded and multiview modes as title should be managed by host apps. This PR makes the title optional, hence if not provided it won't result in the website title being updated. 

[1]: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/title

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] I followed the [breaking change policy] and added [Data Driven Fixes] where supported.
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#overview
[Tree Hygiene]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md
[test-exempt]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md
[Features we expect every widget to implement]: https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/blob/main/docs/contributing/Chat.md
[Data Driven Fixes]: https://github.com/flutter/flutter/blob/main/docs/contributing/Data-driven-Fixes.md

### CODE REVIEW

File: packages/flutter/lib/src/widgets/title.dart [hunk 1]
```diff
@@ -17,13 +17,13 @@ class Title extends StatelessWidget {
   /// [color] and [child] are required arguments.
   Title({
     super.key,
-    this.title = '',
+    this.title,
```
Comment (goderbauer) [inline packages/flutter/lib/src/widgets/title.dart:20]: I think I would leave the Title widget itself unchanged (it's kinda strange that the title widget has a mode where it doesn't set a title) and instead modify the implementation in WidgetsApp to exclude the Title widget if WidgetsApp.title is null.

File: packages/flutter/lib/src/widgets/app.dart [hunk 2]
```diff
@@ -820,7 +820,7 @@ class WidgetsApp extends StatefulWidget {
   ///
   /// To provide a localized title instead, use [onGenerateTitle].
   /// {@endtemplate}
-  final String title;
+  final String? title;
```
Comment (goderbauer) [inline packages/flutter/lib/src/widgets/app.dart:823]: It's odd that WidgetsApp.title and MaterialApp.title / CupertinoApp.title have different defaults (empty string vs null). Those should probably stay in sync.

File: packages/flutter/lib/src/material/app.dart [hunk 3]
```diff
@@ -266,7 +266,7 @@ class MaterialApp extends StatefulWidget {
     this.routerConfig,
     this.backButtonDispatcher,
     this.builder,
-    this.title = '',
+    this.title,
```
Comment (goderbauer) [inline packages/flutter/lib/src/material/app.dart:269]: What does this change in default values mean for Android (the only other platform besides web that uses this)?

> On Android the titles appear above the task manager's app snapshots which are displayed when the user presses the "recent apps" button

Does changing the defaults here changes what's displayed there? If so, we may have to leave the defaults in place and instruct people to pass in null explicitly if they don't want to set a title.

Comment (tugorez) [reply packages/flutter/lib/src/material/app.dart:269]: @reidbaker can this cause any troubles for Android?

File: packages/flutter/lib/src/widgets/app.dart (hunk 2 above)
Comment (tugorez) [reply packages/flutter/lib/src/widgets/app.dart:823]: Oops, seems like I missed the `WidgetsApp.title` ones? Done.

Great catch!

File: packages/flutter/lib/src/widgets/title.dart (hunk 1 above)
Comment (tugorez) [reply packages/flutter/lib/src/widgets/title.dart]: Makes sense. Done!

File: packages/flutter/lib/src/material/app.dart (hunk 3 above)
Comment (tugorez) [reply packages/flutter/lib/src/material/app.dart:269]: Discussed offline with @reidbaker . We're not sure of the implications on Android. Changed the PR to make the behavior changes web only.

**Review by reidbaker — APPROVED** [review]

**Comment by tugorez** [comment]
@yjbanov @Piinks do you happen to know why not instantiating the `Title` widgets leads to `Navigator` not being found in https://github.com/flutter/flutter/blob/master/packages/flutter/test/widgets/app_overrides_test.dart#L79?

**Comment by Piinks** [comment]
> do you happen to know why not instantiating the Title widget leads to Navigator not being found

I do not, maybe @chunhtai has some context?

File: packages/flutter/lib/src/widgets/app.dart [hunk 4]
```diff
@@ -1778,9 +1778,11 @@ class _WidgetsAppState extends State<WidgetsApp> with WidgetsBindingObserver {
           );
         },
       );
+    } else if (widget.title == null && kIsWeb) {
+      title = null;
```
Comment (chunhtai) [inline packages/flutter/lib/src/widgets/app.dart:1782]: you should probably merge this `else if` case to the `if` case, setting this to null will drop the entire subtree

Comment (tugorez) [reply packages/flutter/lib/src/widgets/app.dart:1782]: Great catch!

**Review by chunhtai — APPROVED** [review]
LGTM

**Pull request merged**